src/fork/pty/master/ptsname_r_macos.rs

```diff
@@ -0,0 +1,133 @@
+//! macOS implementation of ptsname_r
+//!
+//! As `ptsname_r()` is not available on macOS, this provides a compatible
+//! implementation using the `TIOCPTYGNAME` ioctl syscall.
+//!
+//! Based on: https://tarq.net/posts/ptsname-on-osx-with-rust/
+
+#[cfg(any(target_os = "macos"))]
+pub unsafe fn ptsname_r(
+    fd: libc::c_int,
+    buf: *mut libc::c_char,
+    buflen: libc::size_t,
+) -> libc::c_int {
+    const IOCTL_BUF_SIZE: usize = 128;
+
+    if buf.is_null() || buflen == 0 {
+        return libc::EINVAL;
+    }
+
+    let mut ioctl_buf: [libc::c_char; IOCTL_BUF_SIZE] = [0; IOCTL_BUF_SIZE];
+
+    if libc::ioctl(fd, libc::TIOCPTYGNAME as libc::c_ulong, &mut ioctl_buf) != 0 {
+        return *libc::__error();
+    }
+
+    let mut len = 0;
+    while len < IOCTL_BUF_SIZE && ioctl_buf[len] != 0 {
+        len += 1;
+    }
+    len += 1;
+
+    if len > buflen {
+        return libc::ERANGE;
+    }
+
+    libc::memcpy(
+        buf as *mut libc::c_void,
+        ioctl_buf.as_ptr() as *const libc::c_void,
+        len,
+    );
+
+    0
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::ffi::CStr;
+
+    #[cfg(any(target_os = "macos"))]
+    #[test]
+    fn test_ptsname_r_retrieves_valid_name() {
+        let master_fd = unsafe { libc::posix_openpt(libc::O_RDWR | libc::O_NOCTTY) };
+        assert!(master_fd >= 0, "Failed to open master PTY");
+
+        unsafe {
```

Comment:
Please add a Saftey comment. I know it's just a test, but having them on every block makes auditing easier.